src/actions/guides/authentication/intro.cr

```diff
@@ -0,0 +1,153 @@
+class Guides::Authentication::Intro < GuideAction
+  guide_route "/authentication"
+
+  def self.title
+    "Authentication Introduction"
+  end
+
+  def markdown
+    <<-MD
+    ## Overview
+
+    When creating a new Lucky project you can choose to generate files for
+    authentication with email and password.
+
+    **If you have an API only Lucky app**, Lucky will generate operations,
+    models, mixins, and actions for signing up and authenticating with an
+    auth token (JWT).
+
+    **If you have a full Lucky app** you will still get API auth, but also
+    actions and pages for signing in, signing out, and resetting your
+    password through the browser.
+
+    ## Retrieving the signed in user
+
+    There are in depth guides on [authentication with HTML pages](#{Guides::Authentication::Browser.path})
+    and [APIs](#{Guides::Authentication::Api.path}), but here
+    are some quick examples of working with the signed in user.
+
+    ### Get the signed in user
+
+    To get the currently signed in user, call `current_user` in your actions.
+    If your action requires sign in, `current_user` will always return a `User`
+    and never `nil`. If sign in is
+    [optional](#{Guides::Authentication::Browser.path(anchor: Guides::Authentication::Browser::ANCHOR_ALLOW_GUESTS)})
+    then  `current_user` may be `nil`.
+
+    In HTML pages using the `MainLayout` you can get the signed in user with
+    `@current_user`.
+
+    ### When current user might be nil
+
+    If the current user might be `nil` by
+    [allowing guests](#{Guides::Authentication::Browser.path(anchor: Guides::Authentication::Browser::ANCHOR_ALLOW_GUESTS)})
+    or [skipping the auth token](#{Guides::Authentication::Api.path(anchor: Guides::Authentication::Api::ANCHOR_OPTIONAL_TOKEN)})
+    you need to use a conditional before using `current_user`
+
+    ```crystal
+    class Hello::Show < BrowserAction
+      # This will allow users that are not signed in
+      include Auth::AllowGuests
+
+      route do
+        user = current_user
+
+        # Check if user is signed in
+        if user
+          plain_text "Hello \#{user.name}"
+        else
+          plain_text "Hello!"
+        end
+      end
+    end
+    ```
+
+    ### Associating database records with the current user
+
+    Let's say we have an `Article` model with a `title : String` and `belongs_to author : User`.
+
+    To create the article and associate it the signed in user, do this:
+
+    ```crystal
+    class Articles::Create < BrowserAction
+      route do
+        # Set 'author_id' to the current_user's id
+        SaveArticle.create!(params, author_id: current_user.id) do |op, article|
+          # ...
+        end
+      end
+    end
+    ```
+
+    ### Filter records by current_user
+
+    Let's say we only want to return articles for the signed in user. We
```

Comment:
"Let's say" can start to sound a bit strange when used multiple times with no preface or followup.

```suggestion
    If we only want to return articles for the signed in user. We
```

Reply:
Good idea!